lib/ProductOpener/APIProductWrite.pm

```diff
@@ -213,9 +213,15 @@ sub update_tags_fields ($request_ref, $product_ref, $field, $add_to_existing_tag
 		);
 	}
 	else {
-		# Generate a comma separated list of tags, so that we can use existing functions to add tags
+		# Generate a comma separated list of tags, so that we can use existing functions to add tags or remove tags
 		my $tags_list = join(',', @$value);
 
+		if ($remove_tags && ref($remove_tags) eq 'ARRAY') {
+			foreach my $tag (@$remove_tags) {
+				remove_tags_from_field($product_ref, $tags_lc, $field, $tag);
+			}
```

Comment:
There is a mistmacht between what you do here and you implementation which expect a list reference (which is the right way I think).

Reply:
I don't understand, are you saying this because of the for loop or because the $tag variable is an element of the reference list?

Reply:
Your code is working but you are doing one call per field, while it could be grouped (or am I missing something ?)

Why not:
```suggestion
			remove_tags_from_field($product_ref, $tags_lc, $field, join(',', @$remove_tags));
			
```

Reply:
Yes, you are right. I did that because it is already handled by the remove_tags_from_field method

Reply:
It is one call per tag to be removed